### ISSUE TITLE
Consider a automatic way to match privileged and hostnetwork constraint 

### ISSUE THREAD

**Issue by resouer** [kubernetes/kubernetes#17039]
Copied from: [this thread](https://groups.google.com/forum/#!topic/kubernetes-dev/VWpCdEF2rgk)

Personally, I'm not strongly request for this,  just feel better if we have.

btw, most of  boolean declarations in Pod are tend to be automatically matched, instead of manually configure node selector , I think.

**Comment by resouer** [comment]
The typical use case is: [daemon set](https://github.com/kubernetes/kubernetes/blob/master/docs/admin/daemons.md)

For example: I want to deploy a glusterfs servers as daemon set. But only part of my clusters are host network && privileged enabled.

Of course it can always be replaced by node-selector

**Comment by davidopp** [comment]
I re-read the email thread and am still confused what you want.

If the Kubelet automatically adds a label indicating the node is host network enabled, and a different label indicating the node is privileged  enabled, (#9044 is relevant) what is the downside to using a label selector on your DaemonSet PodTemplate to say that you want to only schedule on these nodes?

It seems that we need some way for the user to tell the scheduler whether it wants to restrict scheduling to such nodes, and label selectors are the way to do it. What other approach are you advocating?

**Comment by resouer** [comment]
@davidopp Sorry for mis-understanding, I'm not intent to refactor the scheduling (actually, I'd like to talk about scheduling topic, but definitely not in this issue)

What I'm asking for is just automation, not some scheduling optimization. So "If the Kubelet automatically adds a label indicating the node is host network enabled", that will solve my problem. 

I did not notice #9044 before, but it seems a nice place to start. 

I changed the issue title.
